src/main/java/net/dv8tion/jda/core/requests/RestAction.java

```diff
@@ -212,14 +272,70 @@ public RestAction(JDA api, Route.CompiledRoute route, RequestBody data)
      * @param  data
      *         The data that should be sent to the specified route. (can be null)
      */
-    public RestAction(JDA api, Route.CompiledRoute route, JSONObject data)
+    public RestAction(JDA api, Route.CompiledRoute route, RequestBody data, Function<Response, T> successTransformer)
+    {
+        this(api, route, data, successTransformer, DEFAULT_FAILURE_TRANSFORMER);
+    }
+
+    /**
+     * Creates a new RestAction instance
+     *
+     * @param  api
+     *         The current JDA instance
+     * @param  route
+     *         The {@link net.dv8tion.jda.core.requests.Route.CompiledRoute Route.CompiledRoute}
+     *         to be used for rate limit handling
+     * @param  data
+     *         The data that should be sent to the specified route. (can be null)
+     */
+    public RestAction(JDA api, Route.CompiledRoute route, JSONObject data, Function<Response, T> successTransformer)
+    {
+        this(api, route, data == null ? null : RequestBody.create(Requester.MEDIA_TYPE_JSON, data.toString()), successTransformer);
+
+        this.rawData = data;
+    }
+
+    /**
+     * Creates a new RestAction instance
+     *
+     * @param  api
+     *         The current JDA instance
+     * @param  route
+     *         The {@link net.dv8tion.jda.core.requests.Route.CompiledRoute Route.CompiledRoute}
+     *         to be used for rate limit handling
+     * @param  data
+     *         The data that should be sent to the specified route. (can be null)
+     */
+    public RestAction(JDA api, Route.CompiledRoute route, JSONObject data, Function<Response, T> successTransformer, Function<Response, Throwable> failureTransformer)
     {
-        this(api, route, data == null ? null : RequestBody.create(Requester.MEDIA_TYPE_JSON, data.toString()));
+        this(api, route, data == null ? null : RequestBody.create(Requester.MEDIA_TYPE_JSON, data.toString()), successTransformer, failureTransformer);
 
         this.rawData = data;
     }
 
     /**
+     * Creates a new RestAction instance
+     *
+     * @param  api
+     *         The current JDA instance
+     * @param  route
+     *         The {@link net.dv8tion.jda.core.requests.Route.CompiledRoute Route.CompiledRoute}
+     *         to be used for rate limit handling
+     * @param  data
+     *         The data that should be sent to the specified route. (can be null)
+     */
+    public RestAction(JDA api, Route.CompiledRoute route, RequestBody data, Function<Response, T> successTransformer, Function<Response, Throwable> failureTransformer)
+    {
+        Checks.notNull(api, "api");
+
+        this.api = (JDAImpl) api;
+        this.route = route;
+        this.data = data;
+        this.successTransformer = successTransformer;
+        this.failureTransformer = r -> Optional.ofNullable(failureTransformer.apply(r)).orElse(DEFAULT_FAILURE_TRANSFORMER.apply(r));
```

Comment:
What does this do exactly? it assumes that `failureTransformer != null`, then either returns `failureTransformer.apply(r)` or if that is `null` it returns `DEFAULT_FAILURE_TRANSFORMER.apply(r)`?